### PR TITLE
8316462: sun/jvmstat/monitor/MonitoredVm/MonitorVmStartTerminate.java ignores VM flags

### PR DESCRIPTION
**Pull request by lmesnik**
Test fixed to accept vm  flags.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316462](https://bugs.openjdk.org/browse/JDK-8316462): sun/jvmstat/monitor/MonitoredVm/MonitorVmStartTerminate.java ignores VM flags (**Sub-task** - P4)

### Reviewers
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer) ⚠️ Review applies to [da6d3647](https://git.openjdk.org/jdk/pull/16081/files/da6d3647538d25a8a180854e420195ab17b359ce)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [da6d3647](https://git.openjdk.org/jdk/pull/16081/files/da6d3647538d25a8a180854e420195ab17b359ce)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16081/head:pull/16081` \
`$ git checkout pull/16081`

Update a local copy of the PR: \
`$ git checkout pull/16081` \
`$ git pull https://git.openjdk.org/jdk.git pull/16081/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16081`

View PR using the GUI difftool: \
`$ git pr show -t 16081`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16081.diff">https://git.openjdk.org/jdk/pull/16081.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16081#issuecomment-1751393330)

### CODE REVIEW

**Comment by dcubed-ojdk** [comment]
Hmmm... what kind of testing has been done to make sure that we don't
suddenly get a bunch of new failures by allowing this test to accept VM flags?

File: test/jdk/sun/jvmstat/monitor/MonitoredVm/MonitorVmStartTerminate.java [hunk 1]
```diff
@@ -309,9 +309,9 @@ public void terminate() {
         private void executeJava() throws Throwable {
             String className = JavaProcess.class.getName();
             String classPath = System.getProperty("test.classes");
```
Comment (kevinjwalls) [inline test/jdk/sun/jvmstat/monitor/MonitoredVm/MonitorVmStartTerminate.java:311]: this line can be removed also

**Review by kevinjwalls — APPROVED** [review]
Looks good.
So the classPath did not need setting here anyway?  I see createTestJvm ends up adding VM and JAVA options, but I don't see classpath in those.  This change works for me.

Separately I will have a PR for JDK-8209595 which is the timeout failure for this test.

**Review by sspitsyn — APPROVED** [review]
Looks good modulo comment from Kevin.
Thanks,
Serguei

**Comment by lmesnik** [comment]
@dcubed-ojdk I tested this fix by running it with different VM flags.
I see the problems related to 8209595: MonitorVmStartTerminate.java timed out #16077
and going to wait until the fix is integrated to re-test my fix.

File: test/jdk/sun/jvmstat/monitor/MonitoredVm/MonitorVmStartTerminate.java (hunk 1 above)
Comment (lmesnik) [reply test/jdk/sun/jvmstat/monitor/MonitoredVm/MonitorVmStartTerminate.java]: fixed.

**Pull request closed without merging**